c8d/rmi: Don't emit Untagged for dangling images

vvoland opened the pull request:
A dangling image isn't really a tag.

**- What I did**

**- How I did it**

**- How to verify it**
```diff
$ docker rmi 63be88a405ec
-Untagged: moby-dangling@sha256:63be88a405ec007463ceb1b9c5546cd88c9af839e55bb34add6a246962b078f8
Deleted: sha256:63be88a405ec007463ceb1b9c5546cd88c9af839e55bb34add6a246962b078f8
```

**- Description for the changelog**
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->


**- A picture of a cute animal (not mandatory but encouraged)**

vvoland committed:
c8d/rmi: Don't emit Untagged for dangling images

A dangling image isn't really a tag.

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

corhere commented on daemon/containerd/image_delete.go:
```diff
@@ -108,6 +108,9 @@ func (i *ImageService) ImageDelete(ctx context.Context, imageRef string, force,
 			return nil, err
 		}
 
+		if isDanglingImage(img) {
+			return nil, nil
+		}
```
If we can already know beforehand that `img` is a dangling image, why are we still calling `softImageDelete` on it? Is `i.softImageDelete` robust to being passed a dangling image? From a read of the sources, it looks like the answer is no: it will delete the dangling image.

vvoland replied:
This isn't really something that should happen in "normal scenario" as a dangling image should be picked up earlier and wouldn't reach this point at all.
Currently it can only reach here if you use an explicit dangling name (`moby-dangling@digest`), but we don't really want users to actually use this name.
Removed this check and adjusted the earlier code to handle this situation earlier.

vvoland committed:
c8d/rmi: Handle explicit dangling name

This isn't something that user should do, but technically the dangling
images exist in the image store and user can pass its name (`moby-dangling@digest`).
Change it so rmi now recognizes that it's actually a dangling image and
doesn't handle it like a regular tagged image.

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

rumpl commented on daemon/containerd/soft_delete.go:
```diff
@@ -11,6 +11,8 @@ import (
 	"github.com/pkg/errors"
 )
 
+const imageNameDanglingPrefix = "moby-dangling@"
```
For a followup: we should find a new home for this and use the same const in the builder wrapper too

thaJeztah replied:
- created : https://github.com/moby/moby/issues/46613

thaJeztah reviewed (approved):
LGTM
